app/controllers/messages/ViewAllMessagesController.scala

```diff
@@ -36,41 +38,48 @@ class ViewAllMessagesController @Inject()(mcc: MessagesControllerComponents,
                                           val betaAllowList: BetaAllowListAction,
                                           getMessagesService: GetMessagesService,
                                           val view: ViewAllMessages,
-                                          errorHandler: ErrorHandler
-                                         )(implicit val executionContext: ExecutionContext) extends FrontendController(mcc) with AuthActionHelper with I18nSupport {
+                                          errorHandler: ErrorHandler)
+                                         (implicit val executionContext: ExecutionContext)
+  extends FrontendController(mcc) with AuthActionHelper with I18nSupport {
 
-
-  def onPageLoad(ern: String, search: MessagesSearchOptions): Action[AnyContent] =
+  def onPageLoad(ern: String, search: MessagesSearchOptions): Action[AnyContent] = {
     authorisedWithData(ern).async { implicit request =>
+      val session = amendSession(request.session)
```

Comment:
I think this could be moved to the `renderView` function as I don't believe its ever going to be needed on line 51, if we have an index value <=0 then thats caused by the user doing that.